src/lib/integration_tests/tests/platform_orchestrator/reset_ml_feed_cache_test.rs

```diff
@@ -0,0 +1,385 @@
+            encode_one(()).unwrap(),
+        )
+        .map(|reply_payload| {
+            let result: Result<Principal, String> = match reply_payload {
+                WasmResult::Reply(payload) => candid::decode_one(&payload).unwrap(),
+                _ => panic!(
+                    "\n🛑 get_requester_principals_canister_id_create_if_not_exists failed\n"
+                ),
+            };
+            result
+        })
+        .unwrap()
+        .unwrap();
+    println!("res2: {:?}", bob_individual_template_canister_id);
+
+    let dan_individual_template_canister_id = pocket_ic
+        .update_call(
+            user_index_canister_id,
+            dan_principal_id,
+            "get_requester_principals_canister_id_create_if_not_exists",
+            encode_one(()).unwrap(),
+        )
+        .map(|reply_payload| {
+            let result: Result<Principal, String> = match reply_payload {
+                WasmResult::Reply(payload) => candid::decode_one(&payload).unwrap(),
+                _ => panic!(
+                    "\n🛑 get_requester_principals_canister_id_create_if_not_exists failed\n"
+                ),
+            };
+            result
+        })
+        .unwrap()
+        .unwrap();
+    println!("res3: {:?}", dan_individual_template_canister_id);
+
+    pocket_ic.add_cycles(alice_individual_template_canister_id, 2_000_000_000_000);
+    pocket_ic.add_cycles(bob_individual_template_canister_id, 2_000_000_000_000);
+    pocket_ic.add_cycles(dan_individual_template_canister_id, 2_000_000_000_000);
+
+    for i in 0..50 {
+        pocket_ic.tick();
+    }
+
+    // add items into ml cache
+
+    let items = vec![
+        MLFeedCacheItem {
+            post_id: 1 as u64,
+            canister_id: Principal::from_text("ryjl3-tyaaa-aaaaa-aaaba-cai").unwrap(),
+            video_id: "1".to_string(),
+            creator_principal_id: None,
+        },
+        MLFeedCacheItem {
+            post_id: 2 as u64,
+            canister_id: Principal::from_text("ryjl3-tyaaa-aaaaa-aaaba-cai").unwrap(),
+            video_id: "2".to_string(),
+            creator_principal_id: None,
+        },
+        MLFeedCacheItem {
+            post_id: 3 as u64,
+            canister_id: Principal::from_text("ryjl3-tyaaa-aaaaa-aaaba-cai").unwrap(),
+            video_id: "3".to_string(),
+            creator_principal_id: None,
+        },
+    ];
+
+    let res = pocket_ic
+        .update_call(
+            alice_individual_template_canister_id,
+            user_index_canister_id,
+            "update_ml_feed_cache",
+            candid::encode_one(items.clone()).unwrap(),
+        )
+        .map(|res| {
+            let canister_id_result: Result<String, String> = match res {
+                WasmResult::Reply(payload) => candid::decode_one(&payload).unwrap(),
+                _ => panic!("Canister call failed"),
+            };
+            canister_id_result.unwrap()
+        })
+        .unwrap();
```

Comment:
how is update_ml_feed_cache called on mainnet. It is not through platform orchestrator?

Reply:
from off chain agent . super admin